File: src/types/satellite.ts
export interface SatellitePosition {
  id: string;
  name: string;
  latitude: number;
  longitude: number;
  altitude_km: number;
  velocity_kmh: number;
  timestamp: number;
}

export interface OrbitTrack {
  satellite_id: string;
  points: [number, number][];
}

export interface SatelliteData {
  positions: SatellitePosition[];
  orbits: OrbitTrack[];
}

export interface PassPrediction {
  satellite_id: string;
  name: string;
  start_time: number;
  end_time: number;
  max_elevation: number;
  start_azimuth: number;
  is_visible: boolean;
}

File: src-tauri/src/models/satellite.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SatellitePosition {
    pub id: String,
    pub name: String,
    pub latitude: f64,
    pub longitude: f64,
    pub altitude_km: f64,
    pub velocity_kmh: f64,
    pub timestamp: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct OrbitTrack {
    pub satellite_id: String,
    pub points: Vec<[f64; 2]>, // [lat, lon]
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SatelliteData {
    pub positions: Vec<SatellitePosition>,
    pub orbits: Vec<OrbitTrack>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PassPrediction {
    pub satellite_id: String,
    pub name: String,
    pub start_time: i64,   // unix seconds
    pub end_time: i64,
    pub max_elevation: f64, // degrees
    pub start_azimuth: f64, // degrees
    pub is_visible: bool,
}
